client/src/redux/actions/twitterActions.js

```diff
@@ -257,6 +306,7 @@ export const loadUserStreams = ({ streams }) => async (dispatch) => {
         search: stream.search ? stream.search : undefined,
         isLoading: undefined,
         tweets: undefined,
+        metadata: undefined,
       }));
     if (JSON.stringify(streamsArray) !== JSON.stringify(streams)) {
```

Comment:
This is clearly out of the scope of this issue but I feel like I'm missing something here. We should talk about the this thunk next time we meet.

Either way, you could achieve a better performance here by mapping the stream arrays to arrays of stream ids. Then just compare the stream id arrays in the same way. 

Also, is order important here? I. e. should the streams be updated even if the arrays contain the same streams but in a different order?

Reply:
This will be fixed in another PR.
The `streams` array will be a simple array of stream ids and there will be another property `streamsById` that stores every stream. 
Order is important, this is how we know how the user customized his dashboard. 
This action is supposed to sync with the backend and if for some reason streams do not match (order or even missing streams) streams array will be updated.